packages/build/src/plugins_core/edge_functions/index.ts

```diff
@@ -66,19 +65,31 @@ const coreStep = async function ({
       systemLogger: featureFlags.edge_functions_system_logger ? systemLog : undefined,
       internalSrcFolder: internalSrcPath,
     })
+    const metrics = getMetrics(manifest)
 
     systemLog('Edge Functions manifest:', manifest)
+
+    await validateEdgeFunctionsManifest(manifest, featureFlags)
+    return { metrics }
   } catch (error) {
     tagBundlingError(error)
 
     throw error
   }
-
-  await validateEdgeFunctionsManifest({ buildDir, constants: { EDGE_FUNCTIONS_DIST: distDirectory }, featureFlags })
-
-  return {}
 }
 
+const getMetrics = (manifest) => {
```

Comment:
If we created a type for `metrics` and export it from `packages/build/src/core/report_metrics.ts`, we could use it here. It would be a good way to ensure that steps aren't returning metrics with the wrong shape, unsupported type, etc.